File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        #sorted in ascending order, with a target integer, I want to find the index of the target
        #since it is sorted i will search via binary search

        left = 0
        right = len(nums) - 1

        while left <= right:
            middle = left + (right - left) // 2
            if target == nums[middle]:
                return middle
            elif target > nums[middle]:
                left = middle + 1
            else:
                right = middle -1
        
        return -1
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.java
class Solution {
    public int findMin(int[] nums) {
        //techincally everything is still the same order
        //maybe if it is at the bounds then move 

        int smallest = nums[0];
        int left = 0;
        int right = nums.length - 1;

        while(left <= right){
            int middle = left + (right - left)/2;

            if(nums[middle] < smallest){
                smallest = nums[middle];
            }

            //we want to find the smallest
            if(nums[left] <= nums[middle]){
                if(nums[left] < smallest){
                    smallest = nums[left];
                }
                left = middle + 1;
            }
            else{
                right = middle -1; 
            }
        }

        return smallest;


        



    }
}
